harmony/harmonydb/sql/20250423-remove-fee-aggregation.sql

```diff
@@ -0,0 +1,335 @@
+
+    2) Within the same query, we group the rows by (sp_id, reg_seal_proof),
+       chunk them into sub-batches of size up to p_max_batch. For each chunk/batch,
+       we record:
+         - sp_id, reg_seal_proof
+         - sector_nums: an array of sector_number
+         - batch_start_epoch: the MIN start_epoch in that chunk
+         - earliest_ready: the MIN precommit_ready_at in that chunk
+
+    3) We iterate over each batch (in ascending order), and for the first batch that
+       satisfies ANY of:
+         (a) (batch_start_epoch - p_slack_epoch - p_current_height) <= 0
+         (b) (earliest_ready + p_timeout_secs) > NOW()
+       we update those sectors in sectors_sdr_pipeline => task_id_precommit_msg = p_new_task_id,
+       then return the number of rows updated.
+
+    4) If no batch meets the conditions, return 0
+
+  Example usage:
+
+    SELECT poll_start_batch_precommit_msg(
+      randomnessLookBack,    -- p_randomnessLookBack
+      slack_epoch_value,     -- p_slack_epoch
+      current_height_value,  -- p_current_height
+      max_batch_size,        -- p_max_batch
+      new_task_id_value,     -- p_new_task_id
+      timeout_secs           -- p_timeout_secs
+    );
+
+  Returns:
+    The number of sectors updated (i.e., assigned a precommit task) if a batch
+    is found that meets any condition. Otherwise, 0.
+*/
+
+CREATE OR REPLACE FUNCTION poll_start_batch_precommit_msgs(
+    p_randomnessLookBack BIGINT, -- policy.MaxPreCommitRandomnessLookback
+    p_slack_epoch    BIGINT,  -- "Slack" epoch to compare against a sector's start_epoch
+    p_current_height BIGINT,  -- Current on-chain height
+    p_max_batch      INT,     -- Max number of sectors per batch
+    p_new_task_id    BIGINT,  -- Task ID to assign if a batch is chosen
+    p_timeout_secs   INT      -- Timeout in seconds for earliest_ready_at check
+)
+-- We return a TABLE of (updated_count BIGINT, reason TEXT),
+-- but in practice it will yield exactly one row.
+RETURNS TABLE (
+    updated_count BIGINT,
+    reason        TEXT
+)
+LANGUAGE plpgsql
+AS $$
+DECLARE
+batch_rec RECORD;
+    cond_slack   BOOLEAN;
+    cond_timeout BOOLEAN;
+    cond_fee     BOOLEAN;
+BEGIN
+    -- Default outputs if we never find a batch
+    updated_count := 0;
+    reason        := 'NONE';
+    /*
+      We'll do all logic in a single CTE chain, which:
+        (1) Select relevant rows in "sectors_sdr_pipeline".
+        (2) Computes each row's "start_epoch" by coalescing data from
+            sectors_sdr_initial_pieces or fallback.
+        (3) Numbers them per group, forming sub-batches of size p_max_batch.
+        (4) Groups by (sp_id, reg_seal_proof, batch_index).
+        (5) Finally selects sp_id, reg_seal_proof, an array of sector_numbers,
+            plus the min start_epoch and earliest_ready for each batch.
+      Then we loop over those grouped results in ascending order.
+    */
+
+    FOR batch_rec IN
+        WITH locked AS (
```

Comment:
My understanding is that `WITH locked AS` does nothing at all in yugabyte

Reply:
I think you are confusing the table name with "FOR UPDATE;" at the end for actual locking. I can change the table name to avoid confusing.

Reply:
Yaaah it was 11 PM :sweat_smile: